src/hardhat-deploy-ethers.ts

```diff
@@ -0,0 +1,59 @@
+/* Imports: External */
+import { Contract } from 'ethers'
+import { Provider } from '@ethersproject/abstract-provider'
+import { Signer } from '@ethersproject/abstract-signer'
+import { HardhatRuntimeEnvironment } from 'hardhat/types'
+
+export const getDeployedContract = async (
+  hre: HardhatRuntimeEnvironment,
+  name: string,
+  options: {
+    iface?: string
+    signerOrProvider?: Signer | Provider | string
+  } = {}
+): Promise<Contract> => {
+  const deployed = await hre.deployments.get(name)
+
+  await hre.ethers.provider.waitForTransaction(deployed.receipt.transactionHash)
+
+  // Get the correct interface.
+  let iface = new hre.ethers.utils.Interface(deployed.abi)
+  if (options.iface) {
+    const factory = await hre.ethers.getContractFactory(options.iface)
+    iface = factory.interface
+  }
+
+  let signerOrProvider: Signer | Provider = hre.ethers.provider
+  if (options.signerOrProvider) {
+    if (typeof options.signerOrProvider === 'string') {
+      signerOrProvider = hre.ethers.provider.getSigner(options.signerOrProvider)
+    } else {
+      signerOrProvider = options.signerOrProvider
+    }
+  }
+
+  // Temporarily override Object.defineProperty to bypass ether's object protection.
+  const def = Object.defineProperty
+  Object.defineProperty = (obj, propName, prop) => {
+    prop.writable = true
+    return def(obj, propName, prop)
+  }
+
+  const contract = new Contract(deployed.address, iface, signerOrProvider)
+
+  // Now reset Object.defineProperty
+  Object.defineProperty = def
+
+  for (const fnName of Object.keys(contract.functions)) {
+    const fn = contract[fnName].bind(contract)
+    ;(contract as any)[fnName] = async (...args: any) => {
+      const result = await fn(...args)
+      if (typeof result === 'object' && typeof result.wait === 'function') {
+        await result.wait()
```

Comment:
Call `.wait()` here so we don't have to do it every single time we want to interact with a contract.

Reply:
I don't think this is a great change because it means that we'll have to "block" on the address manager registration being mined each time, resulting in slow deployments. OK for now, but let's try to remove in the future?